packages/react-input/src/components/Input/renderInput.tsx

```diff
@@ -1,18 +1,25 @@
 import * as React from 'react';
-import { getSlotsCompat } from '@fluentui/react-utilities';
-import { InputState } from './Input.types';
+import { getSlots } from '@fluentui/react-utilities';
+import { InputSlots, InputState } from './Input.types';
 import { inputShorthandProps } from './useInput';
 
 /**
  * Render the final JSX of Input
  */
 export const renderInput = (state: InputState) => {
-  const { slots, slotProps } = getSlotsCompat(state, inputShorthandProps);
-
+  const { slots, slotProps } = getSlots<InputSlots>(state, inputShorthandProps);
+  delete slotProps.input.children;
+  delete slotProps.inputWrapper.children;
```

Comment:
These `deletes` seem a little fishy. You should throw an error if you find unsupported children instead. Or even better, fix the typings if children are not supported (Omit 'children')

Reply:
Right now the slots would get null rendered if they didn't have children (there's a comment in `useInput`). @bsunderhus has an RFC about how to handle that better.

Reply:
An RFC #18949 that would love more reviews 😉 @behowell